packages/cascara/src/ui/Table/TechnicalDesign.mdx

```diff
@@ -0,0 +1,138 @@
+| onCustomAction | `(String: actionType, Array[String]: selectedItems) => {}` | Fired when a custom action has been applied to one or more rows |
+
+### Config
+
+There will be use cases where features like multi-selection, CRUD, custom actions, etc. won't be required at all, they must be disabled by thefault but easily activated. It makes sense to have boolean props that controll such features.
+
+Another important aspect of configuration is the definition of columns, hence a complete section is devoted to it.
+
+| Name    | Type          | default | Description                                        |
+| ------- | ------------- | ------- | -------------------------------------------------- |
+| Bulk    | Boolean       | `FALSE` | If true, activates the bulk mode (multi-select)    |
+| Columns | Array[Object] | `[]`    | Holds the definitions for each column in the table |
+| Create  | Boolean       | `FALSE` | If true, activates the creation of new items       |
+| Delete  | Boolean       | `FALSE` | If true, activates the deletion of items           |
+| Update  | Boolean       | `FALSE` | If true, activates the edit mode                   |
+
+### Triggers
+
+Custom actions are present in most tables, they are an important aspect of all applications. The nature of the actions depends on the entity they are applied to, hence, an abstract way of defining custom actions is required.
+
+| Attribute | Type     | default      | Description                                                 |
+| --------- | -------- | ------------ | ----------------------------------------------------------- |
+| id        | String   | `''`         | The id of the action, usually the name.                     |
+| callback  | Function | `noop`       | The function to be called when the action button is clicked |
+| trigger   | ANY      | `<Button />` | The element that will trigger the action                    |
+
+### Data
+
+The data passed to the table for rendering purposes.
+
+| Name | Type          | default | Description                                   |
+| ---- | ------------- | ------- | --------------------------------------------- |
+| data | Array[Object] | `[]`    | the tabular data to be displayed by the table |
+
+### Columns
+
+The definition of columns depends on the data, we must make sure we have a well-defined set of data types to allow display and mutation for each one.
+
+| Attribute  | Type                                                                   | default    | Description                                              |
+| ---------- | ---------------------------------------------------------------------- | ---------- | -------------------------------------------------------- |
+| accessor   | String                                                                 | `null`     | The name of the entity attribute that maps to the column |
```

Comment:
- [ ]  Thoughts on just calling this `attribute` instead? I am guessing that would be more simple